lib/store_base_sti_class_for_4_1.rb

```diff
@@ -65,7 +65,12 @@ def join_constraints(foreign_table, foreign_klass, node, join_type, tables, scop
               end
               scope_chain_index += 1
 
-              scope_chain_items.concat [klass.send(:build_default_scope, ActiveRecord::Relation.create(klass, table))].compact
+              default_scope = if klass.method(:build_default_scope).arity == 0
+                klass.send(:build_default_scope)
+              else
+                klass.send(:build_default_scope, ActiveRecord::Relation.create(klass, table))
+              end
+              scope_chain_items.concat [default_scope].compact
```

Comment:
yes, method signature was change in 4.1.x, then added default parameter.
specs failed for that versions,  this is how I discovered